File: Taxi Booking.java
class Solution {
    public static int minimumTime(int N, int cur, int[] pos, int[] time) {
        int min = Integer.MAX_VALUE;
        for(int i=0;i<pos.length;i++){
            int temp = Math.abs(cur-pos[i]) * time[i];
            min = (temp<min) ? temp : min;
        }
        return min;
    }
}
        

File: Taxi Booking.cpp
class Solution{
public:
    int minimumTime(int N,int cur,vector<int> &pos,vector<int> &time){
        int mn = INT_MAX;
        for(int i = 0; i<N; i++){
            int temp = abs( cur - pos[i] ) * time[i];
            mn = min(mn , temp);
        }
        return mn;
    }
};
